### PR TITLE
Natural_gradient

### PR DESCRIPTION
**Pull request by ujohn33**
Added an option for training with natural gradient and gradient clipping as hyperparameter (quanitle and threshold types)

### CODE REVIEW

**Comment by StatMixedML** [comment]
@ujohn33 Thanks for opening the PR.

What's the reason why you closed this again?

**Comment by ujohn33** [comment]
Hi Alexander, 

I have reopened the PR now.
I have added a few more commits since I added and I was not sure if they apply but apparently they do. 

The change was that I sped up the quantile clipping by doing it with only torch avoiding nympy.

Best regards,
Evgeny

**Comment by StatMixedML** [comment]
@ujohn33 Thanks for opening the PR again. Can you please add unit-tests for your changes. Thanks
